cmd/repo-updater/repoupdater/server_test.go

```diff
@@ -891,14 +892,62 @@ func TestServer_ExternalServiceNamespaces(t *testing.T) {
 		ExternalID: "aaaaa",
 	}
 
+	githubExternalServiceConfig := `
+	{
+		"url": "https://github.com",
+		"token": "secret-token",
+		"repos": ["org/repo1", "owner/repo2"]
+	}`
+
+	githubExternalService := types.ExternalService{
+		ID:           1,
+		Kind:         extsvc.KindGitHub,
+		CloudDefault: true,
+		Config:       extsvc.NewUnencryptedConfig(githubExternalServiceConfig),
+	}
+
+	gitlabExternalServiceConfig := `
+	{
+		"url": "https://gitlab.com",
+		"token": "abc",
+		"projectQuery": ["groups/mygroup/projects"]
+	}`
+
+	gitlabExternalService := types.ExternalService{
+		ID:           2,
+		Kind:         extsvc.KindGitLab,
+		CloudDefault: true,
+		Config:       extsvc.NewUnencryptedConfig(gitlabExternalServiceConfig),
+	}
+
+	gitlabRepository := &types.Repo{
+		Name:        "gitlab.com/gitlab-org/gitaly",
+		Description: "Gitaly is a Git RPC service for handling all the git calls made by GitLab",
+		URI:         "gitlab.com/gitlab-org/gitaly",
+		ExternalRepo: api.ExternalRepoSpec{
+			ID:          "2009901",
+			ServiceType: extsvc.TypeGitLab,
+			ServiceID:   "https://gitlab.com/",
+		},
+		Sources: map[string]*types.SourceInfo{
+			gitlabSource.URN(): {
+				ID:       gitlabSource.URN(),
+				CloneURL: "https://gitlab.com/gitlab-org/gitaly.git",
+			},
+		},
+	}
+
+	var idDoesNotExist int64 = 99
```

Comment:
nit: can be a const (or even extracted to the file level)
```suggestion
	const idDoesNotExist int64 = 99
```